resources/helm/v2.1/istio-control/istio-discovery/templates/role.yaml

```diff
@@ -21,11 +21,3 @@ rules:
   resources: ["secrets"]
   # TODO lock this down to istio-ca-cert if not using the DNS cert mesh config
   verbs: ["create", "get", "watch", "list", "update", "delete"]
-
-# For MeshFederation support
-- apiGroups: ["maistra.io"]
-  resources: ["meshfederations", "meshfederations/status", "federationstatuses", "federationstatuses/status", "serviceexports", "serviceexports/status", "serviceimports", "serviceimports/status"]
```

Comment:
@dgn @rcernich I think we should keep this in a separate file and merge it in `role.yaml` during `make gen`?

Reply:
Added mesh federation role as an overly [here](https://github.com/maistra/istio-operator/pull/737/commits/b4109acd053978f9d0fbdc9289b3084ec55010bf).

Reply:
why did this disappear? did we change the approach during the rebase?

Reply:
I believe all the changes for mesh federation have been added to the charts in maistra/istio.  we should not be adding them to the overlays.

Reply:
Right, I have removed the overlay because now after the latest rebase, all the mesh federation changes are coming from maistra/istio.